hack/generate/samples/internal/ansible/advanced_molecule.go

```diff
@@ -0,0 +1,502 @@
+		"args:\n            - \"--metrics-addr=127.0.0.1:8080\"\n            - \"--enable-leader-election\"\n            - \"--leader-election-id=advanced-molecule-operator\"\n            - \"--ansible-args='--vault-password-file /opt/ansible/pwd.yml'\"")
+	pkg.CheckError("adding manager arg", err)
+
+	log.Infof("adding manager env")
+	err = testutils.ReplaceInFile(filepath.Join(ma.ctx.Dir, "config", "manager", "manager.yaml"),
+		"- name: ANSIBLE_GATHERING\n              value: explicit", managerEnv)
+	pkg.CheckError("adding manager env", err)
+
+	log.Infof("adding vaulting args to the proxy auth")
+	err = testutils.ReplaceInFile(filepath.Join(ma.ctx.Dir, "config", "default", "manager_auth_proxy_patch.yaml"),
+		"- \"--leader-election-id=advanced-molecule-operator\"",
+		"- \"--leader-election-id=advanced-molecule-operator\"\n        - \"--ansible-args='--vault-password-file /opt/ansible/pwd.yml'\"")
+	pkg.CheckError("adding vaulting args to the proxy auth", err)
+
+	log.Infof("adding task to not pull image to the config/testing")
+	err = testutils.ReplaceInFile(filepath.Join(ma.ctx.Dir, "config", "testing", "kustomization.yaml"),
+		"- manager_image.yaml", "- manager_image.yaml\n- pull_policy/Never.yaml")
+	pkg.CheckError("adding task to not pull image to the config/testing", err)
+}
+
+func (ma *AdvancedMolecule) addMocksFromTestdata() {
+	log.Infof("adding ansible.cfg")
+	cmd := exec.Command("cp", filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/", "ansible.cfg"), ma.ctx.Dir)
+	_, err := ma.ctx.Run(cmd)
+	pkg.CheckError("adding ansible.cfg", err)
+
+	log.Infof("adding plugins/")
+	cmd = exec.Command("cp", "-r", filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/plugins/"), filepath.Join(ma.ctx.Dir, "plugins/"))
+	_, err = ma.ctx.Run(cmd)
+	pkg.CheckError("adding plugins/", err)
+
+	log.Infof("adding fixture_collection/")
+	cmd = exec.Command("cp", "-r", filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/fixture_collection/"), filepath.Join(ma.ctx.Dir, "fixture_collection/"))
+	_, err = ma.ctx.Run(cmd)
+	pkg.CheckError("adding fixture_collection/", err)
+
+	log.Infof("replacing watches.yaml")
+	cmd = exec.Command("cp", "-r", filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/watches.yaml"), ma.ctx.Dir)
+	_, err = ma.ctx.Run(cmd)
+	pkg.CheckError("replacing watches.yaml", err)
+
+	log.Infof("adding tasks/")
+	cmd = exec.Command("cp", "-r", filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/tasks/"), filepath.Join(ma.ctx.Dir, "molecule/default/"))
+	_, err = ma.ctx.Run(cmd)
+	pkg.CheckError("adding tasks/", err)
+
+	log.Infof("adding secret playbook")
+	cmd = exec.Command("cp", "-r", filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/secret.yml"), filepath.Join(ma.ctx.Dir, "playbooks/secret.yml"))
+	_, err = ma.ctx.Run(cmd)
+	pkg.CheckError("adding secret playbook", err)
+
```

Comment:
This actually seems way more complicated than a shell script. Also why do we use `filepath.Join` if we're going to be using paths already. Particularly, `filepath.Join("../../../hack/generate/samples/internal/ansible/testdata/tasks/")`

Reply:
I will remove the `filepath.Join(`